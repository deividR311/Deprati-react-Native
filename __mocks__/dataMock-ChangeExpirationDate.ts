export const DataMock_ChangeExpirationDate = {
  TEXT_NEWDATE: 'Selecciona el día',
  infoCreditBalance: {
    clientName: 'PATRICIA L. DE UGALDE',
    cardNumber: '1xxxxx6',
    affiliateDate: '2000',
    totalAmount: '5.500,00',
    amountSpent: '724,35',
    aditional: '00',
    availableAmount: '4.775,65',
    pastDueDays: 0,
    nextPayDay: 'Abril 14 de 2023',
    nextAmountToPay: '105,08',
    numeroTarjetaAdicionalDisplay: '14XXX86 00',
    clientFirstName: 'Prueba'
  },
  dataCurrentDate: {
    cuttingDay: 14,
    dueDate: '20230514',
    day: '14'
  },
  dataChangeDate: null,
  selectedDay: '2',
  isAcceptTerms: false,
  showWarning: false,
  isLoadingChangeDate: false,
  isLoadingCurrentDate: false,
  showSucces: false,
  showErrorSucces: false,
  expirationDateContent: {
    legend:
      'No puedes cambiar tu fecha máxima de pago si tienes una deuda pendiente y <b>este cambio se hará efectivo a partir del próximo mes.</b>',
    popup_title:
      '¿Estás seguro de cambiar la fecha máxima de pago de tu crédito De Prati?',
    popup_paragraphs: [
      'Ten en cuenta:',
      '1. Esta acción no se puede deshacer y el cambio se hará efectivo a partir de tu próximo mes.',
      '2. Si haces el cambio de fecha <b>deberás pagar la cuota del mes en curso</b> y este cambio se hará efectivo a partir del próximo mes.'
    ],
    popup_accept_action: 'SI, ESTOY DE ACUERDO',
    popup_cancel_action: 'NO, CANCELAR'
  },
  selectRef: {
    current: {
      reset: jest.fn(),
      openDropdown: jest.fn(),
      closeDropdown: jest.fn()
    }
  },
  setSelectedDay: jest.fn(),
  setIsAcceptTerms: jest.fn(),
  handleChangeDate: jest.fn(),
  setShowWarning: jest.fn(),
  handleCloseSuccess: jest.fn(),
  handleCloseError: jest.fn(),
  handleCloseWarning: jest.fn()
};
