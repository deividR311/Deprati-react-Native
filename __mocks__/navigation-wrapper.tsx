import React from 'react';
import { NavigationContainer } from '@react-navigation/native';
import { createNativeStackNavigator } from '@react-navigation/native-stack';
import { COLORS } from '../src/application/common';
import { ProvideWrapper } from './provider-wrapper';

jest.useFakeTimers();
jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper');

const Stack = createNativeStackNavigator();

export const NavigationWrapper = ({ children, screen, initialParams }: any) => (
  <ProvideWrapper>
    <NavigationContainer>
      <Stack.Navigator
        initialRouteName="default"
        screenOptions={{
          headerShown: true,
          headerShadowVisible: false,
          headerTitleAlign: 'left',
          contentStyle: { backgroundColor: COLORS.WHITE }
        }}>
        {screen && (
          <Stack.Screen
            name="default"
            initialParams={initialParams ?? {}}
            component={screen}
          />
        )}
        {children && <Stack.Screen name="default" component={() => children} />}
      </Stack.Navigator>
    </NavigationContainer>
  </ProvideWrapper>
);
