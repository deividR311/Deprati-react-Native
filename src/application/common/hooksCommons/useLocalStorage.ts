import { remove } from 'lodash';
import { useState, useEffect } from 'react';
import { saveStorage, getStorage } from '../../state-manager/storage/storage';

export const StatusEnum = {
  pending: 'pending',
  success: 'success',
  error: 'error'
};
type StatusEnumType = keyof typeof StatusEnum;

const useLocalStorage = (key, initialValue) => {
  const [status, setStatus] = useState<StatusEnumType>(StatusEnum.pending);
  // State to store our value
  // Pass initial state function to useState so logic is only executed once
  const [storedValue, setStoredValue] = useState(initialValue);

  useEffect(() => {
    getStorage(key, initialValue).then(storage => {
      setStoredValue(storage);
      setStatus(StatusEnum.success);
    });
  }, [key]);

  // Return a wrapped version of useState's setter function that ...
  // ... persists the new value to localStorage.
  const setValue = (value, keyUid?: string) => {
    try {
      // Allow value to be a function so we have same API as useState
      const valueToStore =
        value instanceof Function ? value(storedValue) : value;
      // Save state
      setStoredValue(valueToStore);
      // Save to local storage
      if (typeof saveStorage !== 'undefined') {
        saveStorage(keyUid ?? key, valueToStore);
      }
    } catch (error) {}
  };
  return [storedValue, setValue, status];
};

export default useLocalStorage;
