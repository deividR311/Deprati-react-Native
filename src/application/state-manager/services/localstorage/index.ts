export * from './interface';
export * from './localstorage.slice';
export * from './useLocalStorage';
