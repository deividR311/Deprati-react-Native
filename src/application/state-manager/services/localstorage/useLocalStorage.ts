import { useSelector } from 'react-redux';
import { useAppDispatch } from '../..';
import {
  InitialLocalStorage,
  LocalStorageHook,
  LocalStorageKey,
  LocalStorageParams,
  LocalStorageType
} from './interface';
import {
  localStorageStateSelector,
  readAndStorageInState as _readAndStorageInState,
  save as saveInStorageAndState
} from './localstorage.slice';

export const useLocalStorage = (): LocalStorageHook => {
  const dispatch = useAppDispatch();
  const localStorageData = useSelector(localStorageStateSelector);

  const save = async (values: Partial<LocalStorageType>) => {
    return await dispatch(saveInStorageAndState(values));
  };

  const remove = async (keys: LocalStorageKey[]) => {
    const values = keys.reduce(
      (acc, key) => ({ ...acc, [key]: InitialLocalStorage[key] }),
      {} as LocalStorageParams
    );
    return await dispatch(saveInStorageAndState(values));
  };

  const readAndStorageInState = async () => {
    return await dispatch(_readAndStorageInState());
  };

  const fullClean = async () => {
    return await dispatch(saveInStorageAndState(InitialLocalStorage));
  };

  return {
    localStorageData,
    save,
    remove,
    readAndStorageInState,
    fullClean
  };
};
