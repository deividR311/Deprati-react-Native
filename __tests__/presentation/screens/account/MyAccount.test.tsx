import React from 'react';
import { render, waitFor, fireEvent, act } from '@testing-library/react-native';
import { NavigationWrapper } from '../../../../__mocks__/navigation-wrapper';
import MyAccount from '../../../../src/presentation/screens/account/Account/MyAccount';
import { useLocalStorage } from '../../../../src/application/state-manager/services/localstorage/useLocalStorage';

jest.mock(
  '../../../../src/application/state-manager/services/localstorage/useLocalStorage'
);

describe('MyAccount', () => {
  let screenTest: any;
  beforeEach(() => {
    // mockear useLocalStorage
    useLocalStorage.mockImplementation(() => ({
      save: jest.fn(),
      localStorageData: {
        IS_LOGIN: false // valor inicial de IS_LOGIN
      }
    }));

    screenTest = render(<NavigationWrapper screen={MyAccount} />);
  });

  it('renders correctly in Snapshot', () => {
    expect(screenTest?.toJSON()).toMatchSnapshot();
  });

  it('renders correctly', () => {
    const { getByTestId, getByText } = screenTest;
    expect(getByTestId('btn_sesion')).toBeTruthy();
    expect(getByText('INICIAR SESIÓN')).toBeTruthy();
    //elements menu
    expect(getByText('Mis compras')).toBeTruthy();
    expect(getByText('Mis notificaciones')).toBeTruthy();
    expect(getByText('Mis direcciones')).toBeTruthy();
    expect(getByText('Mis reclamos')).toBeTruthy();
    expect(getByText('Mis devoluciones')).toBeTruthy();
    expect(getByText('Mis datos')).toBeTruthy();
    expect(getByText('Cambio de contraseña')).toBeTruthy();
    expect(getByText('Preguntas frecuentes')).toBeTruthy();
    expect(getByText('Servicio al cliente')).toBeTruthy();
    expect(getByText('Encuentra la tienda más cercana')).toBeTruthy();
  });

  /* it('change button of login', () => {
    const { getByTestId, getByText, queryByTestId, queryByText } = screenTest
    expect(getByTestId('btn_sesion')).toBeTruthy()
    expect(getByText('INICIAR SESIÓN')).toBeTruthy()
    act(() => {
      useLocalStorage.mockImplementation(() => ({
        save: jest.fn(),
        localStorageData: {
          IS_LOGIN: true,
        },
      }))
    })
    waitFor(() => {
      expect(queryByTestId('btn_sesion')).toBeNull()
      expect(queryByText('CERRAR SESIÓN')).toBeTruthy()
    })
  }) */
});
