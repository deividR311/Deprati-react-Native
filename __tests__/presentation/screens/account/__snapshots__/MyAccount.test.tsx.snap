// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyAccount renders correctly in Snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      forwardedRef={[Function]}
      gestureResponseDistance={
        Object {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      stackPresentation="push"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={false}
        hideBackButton={false}
        hideShadow={true}
        largeTitleHideShadow={false}
        title="default"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
      <View
        style={
          Array [
            Object {
              "flex": 1,
            },
            Object {
              "backgroundColor": "rgb(242, 242, 242)",
            },
            Object {
              "backgroundColor": "#ffffff",
            },
          ]
        }
      >
        <RCTSafeAreaView
          emulateUnlessSupported={true}
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <View
            style={
              Object {
                "paddingBottom": 80,
              }
            }
          >
            <View>
              <RCTScrollView
                ListFooterComponent={
                  <React.Fragment>
                    <LinearGradient
                      colors={
                        Array [
                          "#FFFFFF01",
                          "#FFFFFF20",
                          "#0000001A",
                        ]
                      }
                      end={
                        Object {
                          "x": 0.5,
                          "y": 1,
                        }
                      }
                      start={
                        Object {
                          "x": 0.5,
                          "y": 0,
                        }
                      }
                      style={
                        Object {
                          "height": 9,
                          "top": -10,
                          "width": "100%",
                        }
                      }
                    />
                  </React.Fragment>
                }
                ListHeaderComponent={[Function]}
                data={
                  Array [
                    Object {
                      "isLoggedIn": true,
                      "link": "ACCOUNT@ORDERS",
                      "title": "Mis compras",
                    },
                    Object {
                      "isLoggedIn": true,
                      "link": "ACCOUNT@NOTIFICATIONS",
                      "title": "Mis notificaciones",
                    },
                    Object {
                      "isLoggedIn": true,
                      "link": "ACCOUNT@ADDRESS_DELIVERY",
                      "title": "Mis direcciones",
                    },
                    Object {
                      "isLoggedIn": true,
                      "link": "ACCOUNT@SUPPORT_TICKETS",
                      "title": "Mis reclamos",
                    },
                    Object {
                      "isLoggedIn": true,
                      "link": "ACCOUNT@MY_RETURNS",
                      "title": "Mis devoluciones",
                    },
                    Object {
                      "isLoggedIn": true,
                      "link": "ACCOUNT@MY_DATA",
                      "title": "Mis datos",
                    },
                    Object {
                      "actionLink": [Function],
                      "isLoggedIn": true,
                      "link": "",
                      "title": "Cambio de contraseña",
                    },
                    Object {
                      "actionLink": [Function],
                      "isLoggedIn": false,
                      "link": "",
                      "title": "Preguntas frecuentes",
                    },
                    Object {
                      "isLoggedIn": false,
                      "link": "ACCOUNT@COSTUMER_SERVICE",
                      "title": "Servicio al cliente",
                    },
                    Object {
                      "isLoggedIn": false,
                      "link": "ACCOUNT@STORES_LIST",
                      "title": "Encuentra la tienda más cercana",
                    },
                  ]
                }
                getItem={[Function]}
                getItemCount={[Function]}
                keyExtractor={[Function]}
                onContentSizeChange={[Function]}
                onLayout={[Function]}
                onMomentumScrollBegin={[Function]}
                onMomentumScrollEnd={[Function]}
                onScroll={[Function]}
                onScrollBeginDrag={[Function]}
                onScrollEndDrag={[Function]}
                removeClippedSubviews={false}
                renderItem={[Function]}
                scrollEventThrottle={50}
                showsVerticalScrollIndicator={false}
                stickyHeaderIndices={Array []}
                viewabilityConfigCallbackPairs={Array []}
              >
                <View>
                  <View
                    onLayout={[Function]}
                  >
                    <View
                      accessibilityLabel="btn_sesion"
                      accessibilityState={
                        Object {
                          "disabled": false,
                        }
                      }
                      accessible={true}
                      collapsable={false}
                      focusable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Object {
                          "alignItems": "center",
                          "alignSelf": "center",
                          "backgroundColor": "#E31A2C",
                          "borderRadius": 5,
                          "justifyContent": "center",
                          "marginTop": 1,
                          "opacity": 1,
                          "paddingVertical": 10,
                          "width": "95%",
                        }
                      }
                      testID="btn_sesion"
                    >
                      <View
                        style={
                          Object {
                            "flexDirection": "row",
                            "justifyContent": "space-between",
                          }
                        }
                      >
                        <Text
                          style={
                            Array [
                              Object {
                                "color": "white",
                                "fontFamily": undefined,
                                "fontSize": 14,
                                "fontStyle": "normal",
                                "fontWeight": undefined,
                              },
                              Object {},
                              null,
                            ]
                          }
                        >
                          INICIAR SESIÓN
                        </Text>
                      </View>
                    </View>
                    <View
                      style={
                        Array [
                          Object {
                            "backgroundColor": "rgba(0, 0, 0, 0.12)",
                            "height": 0.5,
                          },
                          undefined,
                          Object {
                            "alignSelf": "center",
                            "height": 1,
                            "marginVertical": 15,
                            "width": "95%",
                          },
                        ]
                      }
                    />
                  </View>
                  <View
                    onLayout={[Function]}
                    style={null}
                  >
                    <View
                      accessible={true}
                      collapsable={false}
                      focusable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Object {
                          "opacity": 1,
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "flexDirection": "row",
                            "justifyContent": "space-between",
                            "paddingHorizontal": 24,
                            "paddingVertical": 16,
                          }
                        }
                      >
                        <Text
                          style={
                            Object {
                              "color": "#000000",
                              "fontFamily": "Roboto-Regular",
                              "fontSize": 16,
                            }
                          }
                        >
                          Mis compras
                        </Text>
                        <View>
                          <Text
                            allowFontScaling={false}
                            selectable={false}
                            style={
                              Array [
                                Object {
                                  "color": undefined,
                                  "fontSize": 15,
                                },
                                undefined,
                                Object {
                                  "fontFamily": "Material Icons",
                                  "fontStyle": "normal",
                                  "fontWeight": "normal",
                                },
                                Object {},
                              ]
                            }
                          >
                            
                          </Text>
                        </View>
                      </View>
                    </View>
                  </View>
                  <View
                    onLayout={[Function]}
                    style={null}
                  >
                    <View
                      accessible={true}
                      collapsable={false}
                      focusable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Object {
                          "opacity": 1,
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "flexDirection": "row",
                            "justifyContent": "space-between",
                            "paddingHorizontal": 24,
                            "paddingVertical": 16,
                          }
                        }
                      >
                        <Text
                          style={
                            Object {
                              "color": "#000000",
                              "fontFamily": "Roboto-Regular",
                              "fontSize": 16,
                            }
                          }
                        >
                          Mis notificaciones
                        </Text>
                        <View>
                          <Text
                            allowFontScaling={false}
                            selectable={false}
                            style={
                              Array [
                                Object {
                                  "color": undefined,
                                  "fontSize": 15,
                                },
                                undefined,
                                Object {
                                  "fontFamily": "Material Icons",
                                  "fontStyle": "normal",
                                  "fontWeight": "normal",
                                },
                                Object {},
                              ]
                            }
                          >
                            
                          </Text>
                        </View>
                      </View>
                    </View>
                  </View>
                  <View
                    onLayout={[Function]}
                    style={null}
                  >
                    <View
                      accessible={true}
                      collapsable={false}
                      focusable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Object {
                          "opacity": 1,
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "flexDirection": "row",
                            "justifyContent": "space-between",
                            "paddingHorizontal": 24,
                            "paddingVertical": 16,
                          }
                        }
                      >
                        <Text
                          style={
                            Object {
                              "color": "#000000",
                              "fontFamily": "Roboto-Regular",
                              "fontSize": 16,
                            }
                          }
                        >
                          Mis direcciones
                        </Text>
                        <View>
                          <Text
                            allowFontScaling={false}
                            selectable={false}
                            style={
                              Array [
                                Object {
                                  "color": undefined,
                                  "fontSize": 15,
                                },
                                undefined,
                                Object {
                                  "fontFamily": "Material Icons",
                                  "fontStyle": "normal",
                                  "fontWeight": "normal",
                                },
                                Object {},
                              ]
                            }
                          >
                            
                          </Text>
                        </View>
                      </View>
                    </View>
                  </View>
                  <View
                    onLayout={[Function]}
                    style={null}
                  >
                    <View
                      accessible={true}
                      collapsable={false}
                      focusable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Object {
                          "opacity": 1,
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "flexDirection": "row",
                            "justifyContent": "space-between",
                            "paddingHorizontal": 24,
                            "paddingVertical": 16,
                          }
                        }
                      >
                        <Text
                          style={
                            Object {
                              "color": "#000000",
                              "fontFamily": "Roboto-Regular",
                              "fontSize": 16,
                            }
                          }
                        >
                          Mis reclamos
                        </Text>
                        <View>
                          <Text
                            allowFontScaling={false}
                            selectable={false}
                            style={
                              Array [
                                Object {
                                  "color": undefined,
                                  "fontSize": 15,
                                },
                                undefined,
                                Object {
                                  "fontFamily": "Material Icons",
                                  "fontStyle": "normal",
                                  "fontWeight": "normal",
                                },
                                Object {},
                              ]
                            }
                          >
                            
                          </Text>
                        </View>
                      </View>
                    </View>
                  </View>
                  <View
                    onLayout={[Function]}
                    style={null}
                  >
                    <View
                      accessible={true}
                      collapsable={false}
                      focusable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Object {
                          "opacity": 1,
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "flexDirection": "row",
                            "justifyContent": "space-between",
                            "paddingHorizontal": 24,
                            "paddingVertical": 16,
                          }
                        }
                      >
                        <Text
                          style={
                            Object {
                              "color": "#000000",
                              "fontFamily": "Roboto-Regular",
                              "fontSize": 16,
                            }
                          }
                        >
                          Mis devoluciones
                        </Text>
                        <View>
                          <Text
                            allowFontScaling={false}
                            selectable={false}
                            style={
                              Array [
                                Object {
                                  "color": undefined,
                                  "fontSize": 15,
                                },
                                undefined,
                                Object {
                                  "fontFamily": "Material Icons",
                                  "fontStyle": "normal",
                                  "fontWeight": "normal",
                                },
                                Object {},
                              ]
                            }
                          >
                            
                          </Text>
                        </View>
                      </View>
                    </View>
                  </View>
                  <View
                    onLayout={[Function]}
                    style={null}
                  >
                    <View
                      accessible={true}
                      collapsable={false}
                      focusable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Object {
                          "opacity": 1,
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "flexDirection": "row",
                            "justifyContent": "space-between",
                            "paddingHorizontal": 24,
                            "paddingVertical": 16,
                          }
                        }
                      >
                        <Text
                          style={
                            Object {
                              "color": "#000000",
                              "fontFamily": "Roboto-Regular",
                              "fontSize": 16,
                            }
                          }
                        >
                          Mis datos
                        </Text>
                        <View>
                          <Text
                            allowFontScaling={false}
                            selectable={false}
                            style={
                              Array [
                                Object {
                                  "color": undefined,
                                  "fontSize": 15,
                                },
                                undefined,
                                Object {
                                  "fontFamily": "Material Icons",
                                  "fontStyle": "normal",
                                  "fontWeight": "normal",
                                },
                                Object {},
                              ]
                            }
                          >
                            
                          </Text>
                        </View>
                      </View>
                    </View>
                  </View>
                  <View
                    onLayout={[Function]}
                    style={null}
                  >
                    <View
                      accessible={true}
                      collapsable={false}
                      focusable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Object {
                          "opacity": 1,
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "flexDirection": "row",
                            "justifyContent": "space-between",
                            "paddingHorizontal": 24,
                            "paddingVertical": 16,
                          }
                        }
                      >
                        <Text
                          style={
                            Object {
                              "color": "#000000",
                              "fontFamily": "Roboto-Regular",
                              "fontSize": 16,
                            }
                          }
                        >
                          Cambio de contraseña
                        </Text>
                        <View>
                          <Text
                            allowFontScaling={false}
                            selectable={false}
                            style={
                              Array [
                                Object {
                                  "color": undefined,
                                  "fontSize": 15,
                                },
                                undefined,
                                Object {
                                  "fontFamily": "Material Icons",
                                  "fontStyle": "normal",
                                  "fontWeight": "normal",
                                },
                                Object {},
                              ]
                            }
                          >
                            
                          </Text>
                        </View>
                      </View>
                    </View>
                  </View>
                  <View
                    onLayout={[Function]}
                    style={null}
                  >
                    <View
                      accessible={true}
                      collapsable={false}
                      focusable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Object {
                          "opacity": 1,
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "flexDirection": "row",
                            "justifyContent": "space-between",
                            "paddingHorizontal": 24,
                            "paddingVertical": 16,
                          }
                        }
                      >
                        <Text
                          style={
                            Object {
                              "color": "#000000",
                              "fontFamily": "Roboto-Regular",
                              "fontSize": 16,
                            }
                          }
                        >
                          Preguntas frecuentes
                        </Text>
                        <View>
                          <Text
                            allowFontScaling={false}
                            selectable={false}
                            style={
                              Array [
                                Object {
                                  "color": undefined,
                                  "fontSize": 15,
                                },
                                undefined,
                                Object {
                                  "fontFamily": "Material Icons",
                                  "fontStyle": "normal",
                                  "fontWeight": "normal",
                                },
                                Object {},
                              ]
                            }
                          >
                            
                          </Text>
                        </View>
                      </View>
                    </View>
                  </View>
                  <View
                    onLayout={[Function]}
                    style={null}
                  >
                    <View
                      accessible={true}
                      collapsable={false}
                      focusable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Object {
                          "opacity": 1,
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "flexDirection": "row",
                            "justifyContent": "space-between",
                            "paddingHorizontal": 24,
                            "paddingVertical": 16,
                          }
                        }
                      >
                        <Text
                          style={
                            Object {
                              "color": "#000000",
                              "fontFamily": "Roboto-Regular",
                              "fontSize": 16,
                            }
                          }
                        >
                          Servicio al cliente
                        </Text>
                        <View>
                          <Text
                            allowFontScaling={false}
                            selectable={false}
                            style={
                              Array [
                                Object {
                                  "color": undefined,
                                  "fontSize": 15,
                                },
                                undefined,
                                Object {
                                  "fontFamily": "Material Icons",
                                  "fontStyle": "normal",
                                  "fontWeight": "normal",
                                },
                                Object {},
                              ]
                            }
                          >
                            
                          </Text>
                        </View>
                      </View>
                    </View>
                  </View>
                  <View
                    onLayout={[Function]}
                    style={null}
                  >
                    <View
                      accessible={true}
                      collapsable={false}
                      focusable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Object {
                          "opacity": 1,
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "flexDirection": "row",
                            "justifyContent": "space-between",
                            "paddingHorizontal": 24,
                            "paddingVertical": 16,
                          }
                        }
                      >
                        <Text
                          style={
                            Object {
                              "color": "#000000",
                              "fontFamily": "Roboto-Regular",
                              "fontSize": 16,
                            }
                          }
                        >
                          Encuentra la tienda más cercana
                        </Text>
                        <View>
                          <Text
                            allowFontScaling={false}
                            selectable={false}
                            style={
                              Array [
                                Object {
                                  "color": undefined,
                                  "fontSize": 15,
                                },
                                undefined,
                                Object {
                                  "fontFamily": "Material Icons",
                                  "fontStyle": "normal",
                                  "fontWeight": "normal",
                                },
                                Object {},
                              ]
                            }
                          >
                            
                          </Text>
                        </View>
                      </View>
                    </View>
                  </View>
                  <View
                    onLayout={[Function]}
                  >
                    <BVLinearGradient
                      colors={
                        Array [
                          33554431,
                          553648127,
                          436207616,
                        ]
                      }
                      endPoint={
                        Object {
                          "x": 0.5,
                          "y": 1,
                        }
                      }
                      locations={null}
                      startPoint={
                        Object {
                          "x": 0.5,
                          "y": 0,
                        }
                      }
                      style={
                        Object {
                          "height": 9,
                          "top": -10,
                          "width": "100%",
                        }
                      }
                    />
                  </View>
                </View>
              </RCTScrollView>
            </View>
            <Modal
              animationType="fade"
              hardwareAccelerated={false}
              onDismiss={[Function]}
              transparent={true}
              visible={false}
            />
            <Modal
              animationType="fade"
              hardwareAccelerated={false}
              onDismiss={[Function]}
              onRequestClose={[Function]}
              transparent={true}
              visible={false}
            />
          </View>
          <Modal
            hardwareAccelerated={false}
            transparent={true}
            visible={false}
          />
        </RCTSafeAreaView>
      </View>
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
